### DISCUSSION TITLE
New Clients Register with BOTH Dex and IdP?

### DISCUSSION THREAD

**Discussion by PeterBParker** [dexidp/dex#3361] [Q&A]
Hello! From my time in the documentation and experimenting with Dex, it seems like I have to add a client (either to the staticClients in the config, or via the gRPC API) to Dex and the IdP that Dex is connected to. Is that right? Why can't a client send a request to Dex and Dex just look it up in the IdP it's configured for?

**Comment by nabokihms** [comment]
I don't think I fully understand the question, but I will try to explain.

Dex is a provider itself. It implements all OIDC provider-required capabilities. At the same time, Dex can be connected to many other providers that use different protocols and standards (other than OIDC).
